PV260_Project/BusinessLayer/DataLoading/CsvFileDownloader.cs

```diff
@@ -0,0 +1,43 @@
+using System.Globalization;
+using System.Net;
+using BusinessLayer.DataLoading.Utils;
+using BusinessLayer.Exceptions;
+using CsvHelper;
+using CsvHelper.Configuration;
+using DataLayer.Models;
+
+namespace BusinessLayer.DataLoading;
+
+public class CsvFileDownloader : IDataDownloader
+{
+    public async Task<DataSet> LoadCsvFile(string path, string delimiter = ",")
+    {
+        var client = new HttpClient();
+        // NOTE: There must be User-Agent specified, otherwise the request will be rejected
+        client.DefaultRequestHeaders.Add("User-Agent", "Chrome/101.0.4951.67");
+        var body = "";
+
+        // File download
+        try
+        {
+            var responseMessage = await client.GetAsync(path);
+            body = await responseMessage.Content.ReadAsStringAsync();
+
+            if (!responseMessage.IsSuccessStatusCode)
+            {
+                throw new WebException($"Responded with {responseMessage.StatusCode}");
+            }
+        }
+        catch (Exception e)
+        {
+            throw new DataLoaderException($"Error while downloading csv file from '{path}'", e);
+        }
+
+        var config = new CsvConfiguration(CultureInfo.InvariantCulture)
```

Comment:
These lines are same as in the `CsvFileLoader`, sounds like its rework time :D Also those two interfaces are same, to I would use just one interface and both loaders can implement same interface. The shared logic can be put in some abstract class which can be inherited by the loaders. The logic inside `CsvOperation` can also be put to this abstract class.

Reply:
We were told at the lectures not to use abstract classes :D. 

Those are not 1:1 the same - in this file, I am using a different type of reader (not reading from a file but from a string). 

However, I am no C# master :D so it is possible to unify somehow those 2 approaches (stringreader and streamreader) I am all in!

Reply:
Wtf who told us about not using abstract classes? :D They are very useful. It is different type of reader but the interface looks the same